crates/zoon/src/element/button.rs

```diff
@@ -8,6 +8,81 @@ use std::marker::PhantomData;
 
 make_flags!(Label, OnPress);
 
+/// Create a customizable Button for your web page.
+/// The button is actually built using a div to avoid default behaviors and
+/// styling from regular Html buttons.
+/// You can create a new button by using its constructor and chain styling.
+///
+/// # Examples
+/// ```no_run
+/// use zoon::*;
+/// let button = Button::new().s(Align::center()).s(Padding::all(5));
+/// ```
+/// You can also create your button with specific events as well and update the styling the way you need by using [signals](https://crates.io/crates/futures-signals)
+///
+/// 1 - Create a signal for local state management.
+///
+/// 2 - The variable **`hovered`** gets updated when the user hovers the button
+///
+/// 3 - The variable **`hovered_signal`** is the notification send by signal
+/// that will actually determine the background color for the button.
+///
+/// # Example
+/// ```no_run
+/// use zoon::{named_color::*, *};
+/// let (hovered, hovered_signal) = Mutable::new_and_signal(false);
+///
+/// let button = Button::new().s(Background::new()
+///     .on_hovered_change(move |is_hovered| hovered.set_neq(is_hovered))
+///     .color_signal(hovered_signal.map_bool_signal(|| GREEN_7, || GREEN_8)));
+/// ```
+/// Many options are available on a button.
+/// Most css styling properties are available with their own struct.
+/// For example, the css property **`with`** become **`With`**.
+/// Css colors are available with macros as well.
```

Comment:
I'm not sure how much we should write about CSS properties because many of them don't exist in Zoon or have other names or syntax. For example there is no `border-radius` but `RoundedCorners` because the original name doesn't make sense -  `border-radius` modifies the element shape and works even if there is no border...
Also perhaps in the future we will use Zoon to write desktop/mobile native apps where there isn't any CSS - then the docs could become a bit confusing.

> For example, the css property **`with`** become **`With`**.
I don't understand this sentence :smile:

Reply:
Well I try to give information to people who knows regular css properties and how to use them in the Zoon Context. So I should rephrase it probably. I understand your sentence so, so yes I need to rephrase.

Reply:
I removed references to css because I understand that Zoon will be used in a context where css does not exist. So the description is now much simpler.